pkg/codegen/go/gen_program_utils.go

```diff
@@ -54,3 +54,32 @@ func (p *promptToInputArrayHelper) getPromptItemType() string {
 func (p *promptToInputArrayHelper) getInputItemType() string {
 	return strings.TrimSuffix(p.destType, "Array")
 }
+
+// Provides code for a method which will be placed in the program preamble if deemed
+// necessary. Because many tasks in Go such as reading a file require extensive error
+// handling, it is much prettier to encapsulate that error handling boilerplate as its
+// own function in the preamble.
+func getHelperMethodIfNeeded(functionName string) (string, bool) {
+	var methodBody string
+
+	switch functionName {
+	case "readFile":
+		methodBody =
+			`func readFileOrPanic(path string) pulumi.StringPtrInput {
+				if fileData, err := ioutil.ReadFile(path); err == nil {
+					return pulumi.String(string(fileData[:]))
+				} else {
+					panic(err.Error())
+				}
+			}`
+	case "sha1":
+		methodBody =
+			`func sha1(input string) pulumi.StringPtrInput {
+				inputBytes := []byte(input)
+				return pulumi.String(fmt.Sprintf("%x", sha1.Sum(inputBytes)))
```

Comment:
I don't think that this does what you want it to do:
- This won't compile, since `sha1` in `sha1.Sum` binds to `func sha1`. See https://play.golang.org/p/2ZCM_uilVdp for an example of the error here.
- `fmt.Sprintf("%x", someBytes)` won't format the bytes as a hex string--you need to use [`encoding/hex.EncodeToString](https://pkg.go.dev/encoding/hex#EncodeToString) to do that.

The `sha1Hash` function in this code should do what you want: https://play.golang.org/p/GF-55iRvw00